LoopKitHostedTests/GlucoseStoreTests.swift

```diff
@@ -277,6 +277,47 @@ class GlucoseStoreTests: PersistenceControllerTestCase, GlucoseStoreDelegate {
         XCTAssertEqual([sample1.quantitySample.description], hkobjects.map { $0.description })
     }
     
+    func testGetGlucoseSamplesDeniedHealthKitStorage() {
+        glucoseStore.healthKitStorageDelay = .minutes(5)
+        // Note error should not cause failure
+        healthStore.saveError = Error.arbitrary
+        healthStore.authorizationStatus = .sharingDenied
+        var hkobjects = [HKObject]()
+        healthStore.setSaveHandler { o, _, _ in hkobjects = o }
+        let addGlucoseSamplesCompletion = expectation(description: "addGlucoseSamples")
+        glucoseStore.addGlucoseSamples([sample1, sample2, sample3]) { result in
+            switch result {
+            case .failure(let error):
+                XCTFail("Unexpected failure: \(error)")
+            case .success(let samples):
+                XCTAssertEqual(samples.count, 3)
+            }
+            addGlucoseSamplesCompletion.fulfill()
+        }
+        waitForExpectations(timeout: 10)
+
+        let getGlucoseSamples1Completion = expectation(description: "getGlucoseSamples1")
+        glucoseStore.getGlucoseSamples() { result in
+            switch result {
+            case .failure(let error):
+                XCTFail("Unexpected failure: \(error)")
+            case .success(let samples):
+                XCTAssertEqual(samples.count, 3)
+                // HealthKit storage is denied, so all UUIDs are nil
+                XCTAssertNil(samples[0].uuid)
+                assertEqualSamples(samples[0], self.sample1)
+                XCTAssertNil(samples[1].uuid)
+                assertEqualSamples(samples[1], self.sample3)
+                XCTAssertNil(samples[2].uuid)
```

Comment:
[nit] Consider asserting that `healthKitEligibleDate` is also `nil` for each sample here.

Reply:
Actually, no matter what, `uuid` will be `nil` since, even if sharing wasn't denied, the delay is set for 5 minutes.

Consider two tests then. One for sharing denied with delay of 0 and one for sharing allowed with 5 minute delay.

Reply:
I added a test with a delay of 0 and which changes the sharing authorization between samples (a test I thought of writing after putting up the PR).  Will that cover your concern?

Reply:
Yes, should be good.